pkg/model/external_access.go

```diff
@@ -0,0 +1,69 @@
+package model
+
+import (
+	"github.com/golang/glog"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awstasks"
+	"strconv"
+)
+
+// ExternalAccessModelBuilder configures security group rules for external access
+// (SSHAccess, APIAccess)
+type ExternalAccessModelBuilder struct {
+	*KopsModelContext
+}
+
+var _ fi.ModelBuilder = &ExternalAccessModelBuilder{}
+
+func (b *ExternalAccessModelBuilder) Build(c *fi.ModelBuilderContext) error {
+	if len(b.Cluster.Spec.APIAccess) == 0 {
+		glog.Warningf("APIAccess is empty")
+	}
+
+	if len(b.Cluster.Spec.SSHAccess) == 0 {
+		glog.Warningf("SSHAccess is empty")
+	}
+
+	// SSH is open to AdminCIDR set
+	if b.Cluster.IsTopologyPublic() {
```

Comment:
yes yes indeed - this is far from a deal breaker, and more exciting than anything.. The PRs coming through on this code are going to do just fine - no refactor needed :)